cmake/FindFluidsynth.cmake

```diff
@@ -0,0 +1,65 @@
+#[=======================================================================[.rst:
+FindFluidsynth
+-------
+
+Finds the fluidsynth library.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This will define the following variables:
+
+``Fluidsynth_FOUND``
+  True if the system has the fluidsynth library.
+``Fluidsynth_INCLUDE_DIRS``
+  Include directories needed to use fluidsynth.
+``Fluidsynth_LIBRARIES``
+  Libraries needed to link to fluidsynth.
+``Fluidsynth_VERSION``
+  The version of the fluidsynth library which was found.
+
+Cache Variables
+^^^^^^^^^^^^^^^
+
+The following cache variables may also be set:
+
+``Fluidsynth_INCLUDE_DIR``
+  The directory containing ``fluidsynth.h``.
+``Fluidsynth_LIBRARY``
+  The path to the fluidsynth library.
+
+#]=======================================================================]
+
+find_package(PkgConfig QUIET)
+pkg_check_modules(PC_Fluidsynth QUIET fluidsynth)
+
+find_path(Fluidsynth_INCLUDE_DIR
```

Comment:
Either specify IMPORTED_TARGET and just use the the target or use the PC_Fluidsynth_LINK_LIBRARIES variable for the libs (since 3.12). The PC_Fluidsynth_INCLUDE_DIRS variable cn be used directly.